### PR TITLE
Fixes #57: `attr-for-selected` now always expects hyphenated names.

### LINKED ISSUE #57: The value of attr-for-selected must be camel-cased, not hyphenated?
While working with a ``paper-dropdown-menu``, ``paper-menu``, and ``paper-item``, I couldn't understand why some of the dropdowns weren't displaying the initially selected item or allowing me to change the selection.  I finally discovered through lots of trial and error that ``attr-for-selected`` expects a camelcase value.  Declaring ``attr-for-selected="foo-bar"`` breaks things as described above.  Changing that to ``attr-for-selected="fooBar"`` makes things work as expected.  I suggest that iron-selector be fixed to allow things to work either way, or update the documentation to make it clear that camelcase is required.

### PR DESCRIPTION
**Pull request by bicknellr**
Continuation of #101. (Fixes #57.)

### CODE REVIEW

**Comment by notwaldorf** [comment]
Is this a breaking change? Like, did it use to work with both and now it doesn't?

**Comment by bicknellr** [comment]
Hmm. Previously, the behavior for finding the item's key was pretty much `item[attrForSelected] || item.getAttribute(attrForSelected)`. You could reference attributes or properties, but not both, because of the different cases. If I'm reading [`dashToCamelCase`'s implementation](https://github.com/Polymer/polymer/blob/6c5afe5/src/lib/case-map.html#L20) right, it shouldn't actually change the behavior for passing camel case strings since `dashToCamelCase` won't actually modify it. So, this change actually handles more cases but adds (unenforced) restrictions to the API. <- I'd say minor version bump for this reason, even though this should still work in all the cases we had before without changes.

p.s. I just noticed `dashToCamelCase` does its own caching, but it has to do a lookup rather than just one comparison; do you think this makes `_lastAttrForSelected`/`_lastPropertyNameForSelected` stuff here overkill?

**Comment by notwaldorf** [comment]
@bicknellr Yeaaah i'd say lookups are free, so maybe keep it cleaner and don't do your own caching?

**Comment by bicknellr** [comment]
Ok, I removed the caching, so it's just a call to `dashToCamelCase` now.

File: iron-selectable.html [hunk 1]
```diff
@@ -46,8 +46,11 @@
     properties: {
 
       /**
-       * If you want to use the attribute value of an element for `selected` instead of the index,
-       * set this to the name of the attribute.
+       * If you want to use an attribute value or property of an element for
+       * `selected` instead of the index, set this to the name of the attribute
+       * or property. The attribute or property name **must be given in
```
Comment (notwaldorf) [inline iron-selectable.html]: I think the comment is a bit of a lie :) You can still use camel case, right? Otherwise it would be a breaking change?

**Comment by notwaldorf** [comment]
LGTM if the comment becomes more of a warning/advice :)

File: iron-selectable.html (hunk 1 above)
Comment (bicknellr) [inline iron-selectable.html]: Just edited this; how does it sound now?

**Comment by notwaldorf** [comment]
👌

**Pull request merged**